pints/_mcmc/_monomial_gamma_hmc.py

```diff
@@ -0,0 +1,507 @@
+#
+# Monomial gamma HMC MCMC method
+#
+# This file is part of PINTS.
+#  Copyright (c) 2017-2019, University of Oxford.
+#  For licensing information, see the LICENSE file distributed with the PINTS
+#  software package.
+#
+from __future__ import absolute_import, division
+from __future__ import print_function, unicode_literals
+import pints
+import numpy as np
+from scipy import integrate
+from scipy import interpolate
+
+
+class MonomialGammaHMCMCMC(pints.SingleChainMCMC):
+    """
+    *Extends:* :class:`SingleChainMCMC`
+
+    Implements Monomial Gamma HMC as described in [1] - a generalisation
+    of HMC as described in [2] - involving a non-physical kinetic energy term.
+
+    Uses a physical analogy of a particle moving across a landscape under
+    Hamiltonian dynamics to aid efficient exploration of parameter space.
+    Introduces an auxilary variable -- the momentum (``p_i``) of a particle
+    moving in dimension ``i`` of negative log posterior space -- which
+    supplements the position (``q_i``) of the particle in parameter space. The
+    particle's motion is dictated by solutions to Hamilton's equations,
+
+        dq_i/dt =   partial_d H/partial_d p_i,
+        dp_i/dt = - partial_d H/partial_d q_i.
+
+    The Hamiltonian is given by,
+
+        H(q,p) =       U(q)       +        KE(p)
+               = -log(p(q|X)p(q)) +
+                 Sigma_i=1^d (-g(p_i) + (2/c) * log(1 + e^(cg(p))))
+
+    where ``d`` is the dimensionality of model and ``m_i`` is the 'mass' given
+    to each particle (often chosen to be 1 as default). Note the KE term
+    is the 'soft' version described in [1].
```

Comment:
Maybe mention meaning of U and KE abbrev.?

Reply:
Done